### PR TITLE
Ratelimit emitters

### PR DESCRIPTION
**Pull request by kdorosh**
Ratelimit emitters to once per second so Sync loops acting on Snapshot() have time to finish before being cancelled.

### CODE REVIEW

File: pkg/code-generator/codegen/templates/snapshot_emitter_template.go [hunk 1]
```diff
@@ -275,7 +278,7 @@ func (c *{{ lower_camel .GoName }}Emitter) Snapshots(watchNamespaces []string, o
 			if err != nil {
 				contextutils.LoggerFrom(ctx).Panicw("error while hashing, this should never happen", zap.Error(err))
 			}
-			if previousHash == currentHash {
+			if !needsSync && previousHash == currentHash {
```
Comment (nfuden) [inline pkg/code-generator/codegen/templates/snapshot_emitter_template.go:281]: When our current and previous hash are the same why would we need to sync? can we add a comment on that

Comment (nfuden) [reply pkg/code-generator/codegen/templates/snapshot_emitter_template.go:281]: Nevermind I see its due to the forceemit change

**Comment by kdorosh** [comment]
still want to explore removing the force emitter altogether and making the 1 second rate limiting timeout configurable

**Comment by nfuden** [comment]
Instead of rate limiting what if we were to keep a queue of needed updates up to some max channel length. Perhaps with rate limiting configurable as a extreme fallback?

**Comment by kdorosh** [comment]
might still be a good idea in the long term. will close for now since i don't suspect it was the root cause of https://github.com/solo-io/gloo/issues/6599

**Comment by kdorosh** [comment]
related fear https://github.com/solo-io/gloo-mesh-enterprise/issues/5107 i still think this is a good idea to merge

but won't fight the team if there's strong objections

**Pull request closed without merging**